HatemHashlamon/task_manager.py

```diff
@@ -0,0 +1,119 @@
+from typing import List
+from task import *
+from utils import *
+
+# List to store all tasks (including urgent tasks)
+tasks: List[Task] = []
+
+def display_menu():
+    print("\nTask Manager")
+    print("1. Add Task")
+    print("2. View Tasks")
+    print("3. Update Task")
+    print("4. Delete Task")
+    print("5. Exit")
+
+def add_task():
+    try:
+        task_id = len(tasks) + 1
+        title = get_valid_input("Enter task title: ")
+        description = get_valid_input("Enter task description: ")
+        due_date = get_valid_input("Enter due date (YYYY-MM-DD): ")
+        validate_date(due_date)  
+        status = get_valid_input("Enter task status (InProgress/Completed): ")
+        validate_status(status) 
+
+        is_urgent = get_valid_input("Is this task urgent? (yes/no): ").strip().lower()
+        if is_urgent.lower() in ["yes", "y"]:
+            priority = get_valid_input("Enter priority (High/Medium/Low): ")
+            validate_priority(priority)  
+            task = UrgentTask(task_id, title, description, due_date, status, priority)
+        else:
+            task = Task(task_id, title, description, due_date, status)
+        
+        tasks.append(task)
+        print("Task added.")
+    except ValueError as e:
+        print(e)
+
+
+def view_tasks():
+    if not tasks:
+        print("No tasks available.")
+        return
+
+    regular_tasks = [task for task in tasks if not isinstance(task, UrgentTask)]
```

Comment:
If you want to use a different list for each type, you should use it consistently throughout the code. If not, there's no need to split it this way; you can just check it in the next for loop.